db/queries/base.py

```diff
@@ -1,203 +1,145 @@
-from collections import namedtuple
-
-from sqlalchemy import select, join, Column as SAColumn, Table as SATable
-
-from django.utils.functional import cached_property
+from sqlalchemy import MetaData, select
 
 from db.records.operations import select as records_select
 from db.columns.base import MathesarColumn
+from db.columns.operations.select import get_column_name_from_attnum
+from db.tables.operations.select import reflect_table_from_oid
 from db.transforms.operations.apply import apply_transformations
 
 
 class DBQuery:
     def __init__(
-        self,
-        base_table,
-        initial_columns,
-        transformations=None,
-        name=None
+            self,
+            base_table_oid,
+            initial_columns,
+            engine,
+            transformations=None,
+            name=None,
     ):
-        assert isinstance(base_table, SATable)
-        self.base_table = base_table
+        self.base_table_oid = base_table_oid
         for initial_col in initial_columns:
             assert isinstance(initial_col, InitialColumn)
         self.initial_columns = initial_columns
+        self.engine = engine
         self.transformations = transformations
         self.name = name
 
     # mirrors a method in db.records.operations.select
     def get_records(self, **kwargs):
-        # NOTE how through this method you can perform a second batch of transformations.
-        # this reflects fact that we can form a query, and then apply temporary transforms on it,
-        # like how you can apply temporary transforms to a table when in a table view.
+        # NOTE how through this method you can perform a second batch of
+        # transformations.  this reflects fact that we can form a query, and
+        # then apply temporary transforms on it, like how you can apply
+        # temporary transforms to a table when in a table view.
```

Comment:
I noticed that you reformatted comment lines to something like 80 characters. I've been using 100 character lines for comment blocks (and 80 for code). I thought I got that from PEP-8, but a quick glance didn't reveal such a recommendation there. Should I format comments to 80 chars?

Reply:
I don't think it matters. Emacs does this automatically when I edit functions under certain circumstances, and I didn't take the time to undo those changes. Generally, I don't think this sort of formatting question is for humans, so I kind of let my editor do what it wants.